wintaser/MainMenu.cpp

```diff
@@ -63,422 +81,439 @@ void Build_Main_Menu(HMENU& MainMenu, HWND hWnd)
+	MyInsertMenuByID(FilesChangeState, i++, Flags, ID_FILES_NEXTSTATE, "Next State", "", "Next State");
+	InsertMenu(FilesChangeState, i++, MF_SEPARATOR, NULL, NULL);
+	for(j = 0; j < 10; j++)
+	{
+		sprintf(Str_Tmp ,"Set &%d", (j+1)%10);
+		MyInsertMenuByID(FilesChangeState, i++, Flags | (Current_State == ((j+1)%10) ? MF_CHECKED : MF_UNCHECKED), ID_FILES_SETSTATE_1 + j, Str_Tmp, "", Str_Tmp);
+	}
+
+	MyInsertMenuByID(FilesSaveState, i++, Flags, ID_FILES_SAVESTATE, "Save State", "\tF5", "Quick &Save");
+	MyInsertMenuByID(FilesSaveState, i++, Flags, ID_FILES_SAVESTATEAS, "Save State as", "\tShift+F5", "&Save State as...");
+	InsertMenu(FilesSaveState, i++, MF_SEPARATOR, NULL, NULL);
+	for(j = 0; j < 10; j++)
+	{
+		sprintf(Str_Tmp ,"Save &%d", (j+1)%10);
+		MyInsertMenuByID(FilesSaveState, i++, Flags, ID_FILES_SAVESTATE_1 + j, Str_Tmp, "", Str_Tmp);
+	}
+
+	MyInsertMenuByID(FilesLoadState, i++, Flags, ID_FILES_LOADSTATE, "Load State", "\tF8", "Quick &Load");
+	MyInsertMenuByID(FilesLoadState, i++, Flags, ID_FILES_LOADSTATEAS, "Load State as", "\tShift+F8", "&Load State...");
+	InsertMenu(FilesLoadState, i++, MF_SEPARATOR, NULL, NULL);
+	for(j = 0; j < 10; j++)
+	{
+		sprintf(Str_Tmp ,"Load &%d", (j+1)%10);
+		MyInsertMenuByID(FilesLoadState, i++, Flags, ID_FILES_LOADSTATE_1 + j, Str_Tmp, "", Str_Tmp);
+	}
+
+	// Menu FilesHistory
+	
+	for(i = 0; i < MAX_RECENT_ROMS; i++)
+	{
+		if (strcmp(Recent_Rom[i], ""))
+		{
+			char tmp[1024];
+			switch (Detect_Format(Recent_Rom[i]) >> 1)
+			{
+				default:
+					strcpy(tmp, "[---]    - "); // does not exist anymore
+					break;
+				case 1:
+					strcpy(tmp, "[MD]   - ");
+					break;
+			}
+			Get_Name_From_Path(Recent_Rom[i], Str_Tmp);
+			strcat(tmp, Str_Tmp);
+			// & is an escape sequence in windows menu names, so replace & with &&
+			int len = strlen(tmp);
+			for(int j = 0; j < len && len < 1023; j++)
+				if(tmp[j] == '&')
+					memmove(tmp+j+1, tmp+j, strlen(tmp+j)+1), ++len, ++j;
+			MyInsertMenuByID(FilesHistory, i, Flags, ID_FILES_OPENRECENTROM0 + i, tmp, "", tmp);
+		}
+		else break;
+	}
+#endif
 	
 	// Graphics Menu
 
 	Flags = MF_BYPOSITION | MF_STRING;
 	
 	i = 0;
-
-	//MENU_L(Graphics, i++, Flags | (!localTASflags.forceWindowed ? MF_CHECKED : MF_UNCHECKED) | (started ? MF_GRAYED : 0), ID_GRAPHICS_ALLOWFULLSCREEN, "", "Allow &Fullscreen / Display Mode Changes", "can't change while running");
-	//InsertMenu(Graphics, i++, MF_SEPARATOR, NULL, NULL);
-	MENU_L(Graphics, i++, Flags | (!localTASflags.forceSoftware ? MF_CHECKED : MF_UNCHECKED) | (started ? MF_GRAYED : 0), ID_GRAPHICS_FORCESOFTWARE, "", "&Allow Hardware Acceleration", "can't change while running");
-	MENU_L(Graphics, i++, Flags | MF_POPUP | ((localTASflags.forceSoftware||started) ? MF_GRAYED : 0), (UINT)GraphicsMemory, "", "Surface &Memory", started ? "can't change while running" : "hardware acceleration must be enabled");
+#if 0

```

Comment:
This `#if 0` can be removed, this one seems to be disabled by mistake somewhere...